api/api/V1/Device.ts

```diff
@@ -1100,40 +1110,142 @@ app.post(
       });
-      if (deviceHistoryEntry) {
-        await models.DeviceHistory.update({
+
+      if (!deviceHistoryEntry) {
+        return next(
+          new ClientError(
+            "No device history settings entry found to add mask regions"
+          )
+        );
+      }
+
+      await models.DeviceHistory.update(
+        {
           settings: {
             ...deviceHistoryEntry.settings,
             maskRegions: maskRegions,
-          }
-        }, {
+          },
+        },
+        {
           where: {
             fromDateTime: deviceHistoryEntry.fromDateTime,
             uuid: device.uuid,
             GroupId: device.GroupId,
-          }
-        });
+          },
+        }
+      );
 
-        return successResponse(response, "Mask regions added successfully");
-      } else {
-        return successResponse(
-          response,
-          "No device history settings entry found to add mask regions"
-        );
-      }
+      return successResponse(response, "Mask regions added successfully");
     } catch (e) {
       console.log(e);
+      return next(
+        new UnprocessableError(
+          "An error occurred while processing the request"
+        )
+      );
     }
   }
 );
 
+/**
+ * @api {get} /api/v1/devices/:deviceId/mask-regions Get device mask-regions 
+ * @apiName GetDeviceMaskRegions
+ * @apiGroup Device
+ * @apiParam {Integer} deviceId Id of the device
+ *
+ * @apiDescription Retrieves mask regions for a device from the DeviceHistory table.
+ *
+ * @apiSuccessExample {JSON} device:
+ * {
+ *   "maskRegions": [
+ *     {
+ *       "region": "0",
+ *       "points": [
+ *         { "x": 0.99, "y": 0.66 },
+ *         { "x": 0.80, "y": 0.83 },
+ *         { "x": 0.58, "y": 0.18 }
+ *       ]
+ *     },
+ *     {
+ *       "region": "1",
+ *       "points": [
+ *         { "x": 0.3, "y": 0.1 },
+ *         { "x": 0.5, "y": 0.7 },
+ *         { "x": 0.8, "y": 0.4 }
+ *       ]
+ *     }
+ *   ]
+ * }
+ *
+ * @apiUse V1UserAuthorizationHeader
+ *
+ * @apiUse V1ResponseSuccess
+ * @apiSuccess {apiSuccess::MaskRegionsData}
+ * @apiUse V1ResponseError
+ */
+
+
+app.get(
+  `${apiUrl}/:id/mask-regions`,
+  extractJwtAuthorizedUser,
+  validateFields([
+    idOf(param("id")),
+    query("at-time").isISO8601().toDate().optional(),
+  ]),
+  fetchAuthorizedRequiredDeviceById(param("id")),
+  async (request: Request, response: Response) => {
+    const atTime = request.query["at-time"] as unknown as Date;
+    const device = response.locals.device;
+    const deviceSettings: DeviceHistory | null = await models.DeviceHistory.findOne({
+      where: {
+        uuid: device.uuid,
+        GroupId: device.GroupId,
+        location: { [Op.ne]: null },
+        fromDateTime: { [Op.lte]: atTime },
+      },
+      order: [["fromDateTime", "DESC"]],
+    });
+
+    if (deviceSettings) {
+      return successResponse(response, "Device mask-regions retrieved successfully", deviceSettings.settings.maskRegions);
```

Comment:
We'd usually name the returned object, so it would look more like:
```
return successResponse(response, "Device mask-regions retrieved successfully", { "maskRegions": deviceSettings.settings.maskRegions });
      ```
I think.